include/FilesystemManagement.h

```diff
@@ -0,0 +1,156 @@
+    /// @param NewFilePath The path (including the filename) to where the file should be named.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully moved, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB MoveFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Deletes a file existing on the filesystem.
+    /// @param FilePath The existing path to the file (including the filename) to be deleted.
+    /// @return Returns true if the operation was successful, false if it failed.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB RemoveFile(const StringView FilePath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Symlinks
+
+    /// @brief Creates a symbolic link to a file on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateSymlink(const StringView SymPath, const StringView TargetPath);
+    /// @brief Creates a symbolic link to a directory on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateDirectorySymlink(const StringView SymPath, const StringView TargetPath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Basic Directory Management
+
+    /// @brief Verifies the existence of a folder.
+    /// @param DirectoryPath The path and name of the folder to check for.
+    /// @return Returns true if the folder at the specified location exists, false otherwise.
+    [[nodiscard]]
+    Boole MEZZ_LIB DirectoryExists(const StringView DirectoryPath);
+
+    /// @brief Creates a single new directory.
+    /// @remarks This function will only create the directory specified at the end of the path.
+    /// @param DirectoryPath The path for the newly created directory.
+    /// @return Returns true if the directory was created, false in the case of a non-critical error.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateDirectory(const StringView DirectoryPath);
+    /// @brief Creates all directories that do not exist in the provided path.
+    /// @param DirectoryPath The path for the newly created directory or directories.
+    /// @return Returns true if all directories were created, false in the case of a non-critical error.
```

Comment:
True and False? I think and most of the return documentation needs to be revised. Also, Success is false :(

Reply:
Most of the documentation for the returns has been revised.  

As for our talk about making the enum implicitly convertable to a bool...that doesn't seem possible in the manner we discussed.  We can make utility functions that to the checks for us...which is ick.  Or we can overload operator! and expect the dev to prefix any call to these functions with "!!", which is also ick.  But nothing seemless seems to be possible.  

If you know of another trick, let me know.

Reply:
I knew operators could be overloaded for enum classes, but I didn't know that operator bool was excluded. It also seems like my initial concern was unfounded `if(CreateDirectory("Foo"))` won't build. I propose we go with the operator!, operator==(bool), and operator!=(bool).

I wrote up some experiments and possible implementations: https://coliru.stacked-crooked.com/a/5c78dc13e9da39a0

Reply:
I have some pretty big beef with the "!!" idiom and have a very strong preference to not go with that.  However your proposal for equality and inequality operators with bool seem good.

Reply:
I've implemented operator== and operator!= for the ModifyResult enum.